### PR TITLE
Add brief tutorial and script for creating new Docker image

### PR DESCRIPTION
**Pull request by joshessman-llnl**
The manual intervention to test the build and grab the new host-config isn't ideal - the issue is that `docker cp` doesn't work with wildcards, so from my understanding moving + committing the new host config has to be manual.  

I suppose we could dump the entire repo back into the host (to avoid the wildcard) and grab the host config that way, but IMO that's equally if not more messy than having the user do it, especially for something that's relatively infrequent.

### CODE REVIEW

**Comment by joshessman-llnl** [comment]
LGTM

File: scripts/docker/build_new_image.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+sed -e "s/<VER>/$ver/g" -e "s/<NAME>/$name/g"  -e "s/<MAJ_VER>/$maj_ver/g" dockerfile.in > "$dockerfile_name"
+
+docker build -t "seracllnl/tpls:$tag_name" - < "$dockerfile_name" || exit 1
+
+echo "Build complete, use the generated host-config to test the build process, then press Enter."
+
+read placeholder
+
+docker push seracllnl/tpls:"$tag_name"
```
Comment (jamiebramwell) [inline scripts/docker/build_new_image.sh]: newline at end of file.

**Review by jamiebramwell — APPROVED** [review]
Thanks for documenting this!

File: src/docs/sphinx/docker_info.rst [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+--------------------
+
+The following instructions apply to the creation of a new compiler image.
+
+    1. Start by cloning down the ``serac`` repository.  
+    #. Ensure that an Axom image exists on Dockerhub for the desired compiler.
+       If no corresponding Axom compiler image exists, it should be 
+       created before proceeding.
+    #. Go to the ``scripts/docker`` directory and run ``build_new_image.sh``, passing the compiler
+       name and version, e.g. for Clang 10, run ``./build_new_image.sh clang 10``.  Minor versions can also be specified.
```
Comment (white238) [inline src/docs/sphinx/docker_info.rst]: It's very common to have a minor version, can you provide an example here for that. I assume you mean just 10.0..

**Review by white238 — APPROVED** [review]
This is great thanks @joshessman-llnl

**Comment by joshessman-llnl** [comment]
LGTM

**Pull request merged**